src/cloudWatchLogs/document/logStreamCodeLensProvider.ts

```diff
@@ -62,7 +63,7 @@ export class LogStreamCodeLensProvider implements vscode.CodeLensProvider {
             codelenses.push(oldCodeLense)
         }
         codelenses.push(newerCodelense)
-
+        highlightDocument(this.registry, uri) // This exists here because this is the last change made to the document before rendered.
```

Comment:
This works but only because we forcefully trigger `onDidChangeCodeLenses`. The implication is that we 'own' the document and so always know when the document has changed.

Can you think of a way to achieve the same behavior without `LogStreamCodeLensProvider`? We don't have to do it in this PR, I just want to make sure we understand how the overall design works. 

Hint: it involves `vscode.workspace.onDidChangeTextDocument`

Reply:
I think i figured out what you're hinting at. The idea is to set `highlightDocument` to run every time the document updates.  I still need to run it once in the beginning, but I think i found an approach that works involving `vscode.workspace.onDidChangeTextDocument` and nothing in the `logStreamCodeLensProvider.ts` file.